src/mapgen/mapgen_fractal.cpp

```diff
@@ -103,8 +103,17 @@ void MapgenFractalParams::readParams(const Settings *settings)
 	settings->getS16NoEx("mgfractal_dungeon_ymax",         dungeon_ymax);
 	settings->getU16NoEx("mgfractal_fractal",              fractal);
 	settings->getU16NoEx("mgfractal_iterations",           iterations);
-	settings->getV3FNoEx("mgfractal_scale",                scale);
-	settings->getV3FNoEx("mgfractal_offset",               offset);
+
+    std::optional<v3f> mgfractalScale;
+    if (settings->getV3FNoEx("mgfractal_scale", mgfractalScale) && mgfractalScale.has_value()) {
+        scale = mgfractalScale.value();
```

Comment:
Nitpick:

```suggestion
        scale = *mgfractalScale;
```

whenever you have already ensured that a value is present, use `*optional` instead of `optional.value()`, which includes a (then unnecessary) check for whether a value is present.

Also applies to a couple `.value()` calls below.